trulens_eval/trulens_eval/appui.py

```diff
@@ -11,19 +11,22 @@
 from trulens_eval.utils.json import jsonify_for_ui
 from trulens_eval.utils.serial import Lens
 
-with OptionalImports(messages=REQUIREMENT_NOTEBOOK):
+with OptionalImports(messages=REQUIREMENT_NOTEBOOK) as opt:
     # Here just for the assertion below. Including in a seperate context because
     # auto import organizer might move it below another import and if that other
     # import fails, this name will not be defined to check the assertion below.
-    import ipywidgets
 
-with OptionalImports(messages=REQUIREMENT_NOTEBOOK):
+    # TODO: The optional imports system should not allow this to happen, figure
+    # out what is going wrong.
+    import ipywidgets
     from ipywidgets import widgets
+
+with opt:
     import traitlets
     from traitlets import HasTraits
     from traitlets import Unicode
 
-OptionalImports(messages=REQUIREMENT_NOTEBOOK).assert_installed(ipywidgets)
+opt.assert_installed(ipywidgets).assert_installed(traitlets)
```

Comment:
not following what this change is here, can you explain? why assert install traitlets also?

Reply:
Just in case it is not installed. It should get installed with ipywidgets but in case someone's installation is broken, this might produce a useful message about traitlets.